assets/src/components/detours/diversionPage.tsx

```diff
@@ -157,7 +157,11 @@ export const DiversionPage = ({
                       selectedRoute: route,
                       selectedRoutePatternId: routePattern?.id ?? null,
                     }
-                  : { selectedRoute: null }
+                  : {
+                      routePatterns: undefined,
+                      selectedRoute: route || null,
```

Comment:
suggestion: this will convert `""` to `null`, instead, if we use the `??` operator, it'll only convert `undefined` and `null` to `null`.
```suggestion
                      selectedRoute: route ?? null,
```